include/soa.h

```diff
@@ -147,6 +147,23 @@ class SoA {
     size_ -= num_elements;
   }
 
+  // Copies an element into another element inside the array.
+  void copy(size_t index0, size_t index1) {
+    if (index0 >= size() || index1 >= size()) {
```

Comment:
I like putting the check inside the assert. At runtime people should make sure they know the ranges of their arrays and we can skip the branch here.

Reply:
Actually, ignore this comment. I forgot I used to do it but changed this. Sorry!

Reply:
No worries. I pretty much followed what you did in `swap`